tests/grpc/grpc_test.go

```diff
@@ -227,3 +228,137 @@ func Test_Grpc_Terminate_Recursive(t *testing.T) {
+		OrchestrationStatuses: []protos.OrchestrationStatus{
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_RUNNING,
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_COMPLETED,
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_PENDING,
+		},
+	}
+
+	id, err := grpcClient.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("世界"), api.WithInstanceID(instanceIDs))
+	require.NoError(t, err)
+	// random sleep a time from 0 to 5 seconds, to allow previous orchestration instance to set in different status
+	randomeDuration := time.Duration(rand.Intn(6)) * time.Second
+	time.Sleep(randomeDuration)
+	// schedule again
+	id, err = grpcClient.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("世界"), api.WithInstanceID(id), api.WithOrchestrationReuseOption(ReuseIdOption))
+	require.NoError(t, err)
+	timeoutCtx, cancelTimeout := context.WithTimeout(ctx, 30*time.Second)
+	defer cancelTimeout()
+	metadata, err := grpcClient.WaitForOrchestrationCompletion(timeoutCtx, id, api.WithFetchPayloads(true))
+	require.NoError(t, err)
+	assert.Equal(t, true, metadata.IsComplete())
+	assert.Equal(t, `"Hello, 世界!"`, metadata.SerializedOutput)
+	time.Sleep(1 * time.Second)
+}
+
+func Test_Grpc_ReuseInstanceIDTerminate(t *testing.T) {
+	delayTime := 2 * time.Second
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("SingleActivity", func(ctx *task.OrchestrationContext) (any, error) {
+		var input string
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, err
+		}
+		var output string
+		ctx.CreateTimer(delayTime).Await(nil)
+		err := ctx.CallActivity("SayHello", task.WithActivityInput(input)).Await(&output)
+		return output, err
+	})
+	r.AddActivityN("SayHello", func(ctx task.ActivityContext) (any, error) {
+		var name string
+		if err := ctx.GetInput(&name); err != nil {
+			return nil, err
+		}
+		return fmt.Sprintf("Hello, %s!", name), nil
+	})
+
+	cancelListener := startGrpcListener(t, r)
+	defer cancelListener()
+	instanceIDs := api.InstanceID("TERMINATE_IF_RUNNING_OR_COMPLETED")
+	ReuseIdOption := api.OrchestrationIDReuseOption{
+		CreateOrchestrationAction: protos.CreateOrchestrationAction_TERMINATE,
+		OrchestrationStatuses: []protos.OrchestrationStatus{
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_RUNNING,
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_COMPLETED,
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_PENDING,
+		},
+	}
+
+	id, err := grpcClient.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("世界"), api.WithInstanceID(instanceIDs))
+	require.NoError(t, err)
+	// random sleep a time from 0 to 5 seconds, to allow previous orchestration instance to set in different status
+	randomeDuration := time.Duration(rand.Intn(6)) * time.Second
+	time.Sleep(randomeDuration)
+	// schedule again
+	id, err = grpcClient.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("世界"), api.WithInstanceID(id), api.WithOrchestrationReuseOption(ReuseIdOption))
```

Comment:
Same feedback as with the above test. It doesn't seem like we have a way to actually know whether we successfully terminated the original orchestration since both instances have the exact same output.